zerver/actions/message_edit.py

```diff
@@ -781,25 +782,33 @@ def user_info(um: UserMessage) -> Dict[str, Any]:
                 # important for security reasons; we don't want to
                 # give users a UserTopic row in a stream they cannot
                 # access.  Unmute the topic for such users.
-                do_unmute_topic(muting_user, stream_being_edited, orig_topic_name)
+                do_set_user_topic_visibility_policy(
+                    muting_user,
+                    stream_being_edited,
+                    orig_topic_name,
+                    visibility_policy=UserTopic.VISIBILITY_POLICY_INHERIT,
+                )
```

Comment:
I think it's safe for this project to be a follow-up, since we won't be offering these new values in the UI yet, but the logic in this `for muting_user in get_users_muting_topic` will want to move all `UserTopic` records, not sure those with the `MUTED` state. And the TODO comment above about bulk updates should ideally be fixed while we're at it.

Reply:
(But maybe we should open an issue for addressing this detail -- can you do that?).

Reply:
Opened #24574 

I'm curious about:
> the TODO comment above about bulk updates should ideally be fixed while we're at it.

Maybe, we would need to write separate `bulk_update()` and `bulk_delete()` query, instead of using `do_set_user_topic_visibility_policy`

Reply:
Yeah, I think something like that -- which is why I suggested we do it as a follow-up; seems like it may be complicated enough to get bogged down.